hubspot/tasks.py

```diff
@@ -44,3 +50,37 @@ def sync_line_item_with_hubspot(line_id):
     body = [make_line_item_sync_message(line_id)]
     response = send_hubspot_request("LINE_ITEM", HUBSPOT_SYNC_URL, "PUT", body=body)
     response.raise_for_status()
+
+
+@app.task
+def check_hubspot_api_errors():
+    """Check for and log any errors that occurred since the last time this was run"""
+    offset = 0
+    last_check, _ = HubspotErrorCheck.objects.get_or_create(
+        defaults={"checked_on": now_in_utc()}
+    )
+    last_timestamp = int(last_check.checked_on.timestamp() * 1000)
+    caught_up = False
+    while not caught_up:
+        resp = get_sync_errors(limit=200, offset=offset)
+        resp.raise_for_status()
+        errors = resp.json().get("results", [])
+
+        if not errors:
+            caught_up = True
+        for error in errors:
+            if error.get("errorTimestamp") > last_timestamp:
+                msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
+                    obj_type=error.get("objectType", "N/A"),
+                    obj_id=error.get("integratorObjectId", "N/A"),
+                    details=error.get("details", ""),
+                )
+                log.error(msg)
+            else:
+                caught_up = True
+                break
+        if errors and not caught_up:
+            offset += 200
```

Comment:
If you created a generator that returned sync errors one at a time and took care of the paging/offset on its own (or just updated `get_sync_errors` to do that), I think this whole function could be rewritten like so:

```python
def check_hubspot_api_errors():
    last_check, _ = HubspotErrorCheck.objects.get_or_create(
        defaults={"checked_on": now_in_utc()}
    )
    last_timestamp = int(last_check.checked_on.timestamp() * 1000)
    for error in get_sync_errors():
        if error.get("errorTimestamp") > last_timestamp:
            msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
                obj_type=error.get("objectType", "N/A"),
                obj_id=error.get("integratorObjectId", "N/A"),
                details=error.get("details", ""),
            )
            log.error(msg)
        else:
            break

    last_check.checked_on = now_in_utc()
    last_check.save()
```